### PR TITLE
8 dockerconfigure services ports volumes and deps

### LINKED ISSUE #8: Docker/Configure services, ports, volumes and deps
### Docker compose set up, open ports, volumes & deps

> Build project according to the subject (compose running multi-container Docker application, https://docs.docker.com/compose/)

#### Primary services used as containers : 

- [x] Server service running on **port 8080**
- [x] Find a stable and light base node ( probably node:16-alpine )

> Used to launch the application server

- [x] _client_mobile_ service

> Used to build the mobile application

- [x] A _client_web_ service on **port 8081**

> Used to launch the web client, note that service is depending on the two previous one (https://docs.docker.com/compose/compose-file/#depends_on)

#### Secondary services used as containers : 

- [x] Graphql service running with apolo
> need to read more about this, maybe found image or dockerfile

- [x] Front container
> Using dockerfile to get every elements and package we need to run it perfectly

### PR DESCRIPTION
**Pull request by Bootoyka**
This pr delivers a first version of docker-compose.yml & first version of web_server Dockerfile according to the subject recommandation

close #8

### CODE REVIEW

**Comment by Bootoyka** [comment]
This is more draft than real pr, didn't find how to turn it draft

**Review by EmileEttel — APPROVED** [review]
LGTM

**Review by Cavonstavant — APPROVED** [review]

**Comment by Bootoyka** [comment]
Let me merge this one please to be sure that everything is working good

**Review by majent4 — APPROVED** [review]

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+#      context: ./client
+#      dockerfile: client_mobile.dockerfile
+#    volumes:
+#      - client_data:/data/client
+  apollo:
+    build:
+      context: ./database
+      dockerfile: Dockerfile
+    ports:
+      - "8082:8082"
```
Comment (Cavonstavant) [inline docker-compose.yml:31]:
```suggestion
      - "4000:4000"
```

**Review by Cavonstavant — CHANGES_REQUESTED** [review]

File: server/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
+FROM node:14-alpine
+
+#Create app directory
+RUN mkdir -p /usr/src/app
+WORKDIR /usr/src/app
+
```
Comment (Cavonstavant) [inline server/Dockerfile:6]: don't run things as root on the container

```suggestion
USER www-data
```

**Review by Cavonstavant — CHANGES_REQUESTED** [review]

File: server/Dockerfile (hunk 2 above)
Comment (Cavonstavant) [reply server/Dockerfile:6]: I should note that USER must be put on top of the 3rd line: `RUN mkdir -p /usr/src/app` or the folder will be created as root

**Pull request merged**